### PR TITLE
GEODE-2764: Added checks on the region name

### PR DESCRIPTION
**Pull request by nabarunnag**
* Substring the region name till the last occurance of '.'
	* Check if this substring is a valid region name
	* Continue this till we get valid region name
	* If there are no '.' left to substring on, send region not found exception.

@jhuynh1 @ladyVader @gesterzhou

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/CreateIndexFunction.java [hunk 1]
```diff
@@ -93,6 +91,31 @@ public void execute(FunctionContext context) {
...
+      context.getResultSender().lastResult(new CliFunctionResult(memberId, false, message));
+    } else {
+      XmlEntity xmlEntity =
+          new XmlEntity(CacheXml.REGION, "name", cache.getRegion(regionPath).getName());
+      context.getResultSender().lastResult(new CliFunctionResult(memberId, xmlEntity));
+    }
+  }
+
+  private String getValidRegionName(Cache cache, String regionPath) {
+    while (cache.getRegion(regionPath) == null && regionPath != null) {
```
Comment (jhuynh1) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/CreateIndexFunction.java]: If regionPath is null, will cache.getRegion(regionPath) throw an exception?  Shouldn't the regionPath check for null be first?

Comment (nabarunnag) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/CreateIndexFunction.java]: Its fixed now.
thank you !!

**Comment by nabarunnag** [comment]
@jhuynh1 
I have the changes you mentioned and also added a dunit test to confirm the behaviour

File: geode-wan/src/test/java/org/apache/geode/management/internal/configuration/ClusterConfigurationIndexWithFromClauseDUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  private MemberVM locator = null;
+
+  @Before
+  public void before() throws Exception {
+    locator = lsRule.startLocatorVM(0);
+  }
+
+  @Test
+  @Parameters(method = "getPartitionRegionTypes")
+  public void indexCreatedWithFromClauseMustPersist(RegionShortcut regionShortcut)
```
Comment (jhuynh1) [inline geode-wan/src/test/java/org/apache/geode/management/internal/configuration/ClusterConfigurationIndexWithFromClauseDUnitTest.java]: maybe change this name to describing the .entrySet() that this was trying to solve?
indexCreatedWithEntrySetInFromClauseMustPersist?

**Pull request closed without merging**